vpc/service/long_lived_lock.go

```diff
@@ -19,6 +24,81 @@ const (
 	timeBetweenTryingToAcquireLocks = 15 * time.Second
 )
 
+func (vpcService *vpcService) readLocks(rows *sql.Rows) ([]*vpcapi.Lock, error) {
+	locks := []*vpcapi.Lock{}
+	for rows.Next() {
+
+		lock := vpcapi.Lock{}
+		var heldUntil time.Time
+		err := rows.Scan(&lock.Id, &lock.LockName, &lock.HeldBy, &heldUntil)
+		if err != nil {
+			return nil, err
+		}
+
+		lock.HeldUntil, err = ptypes.TimestampProto(heldUntil)
+		if err != nil {
+			return nil, err
+		}
+
+		locks = append(locks, &lock)
+	}
+
+	return locks, nil
+}
+
+func (vpcService *vpcService) GetLocks(ctx context.Context, empty *empty.Empty) (*vpcapi.GetLocksResponse, error) {
+	rows, err := vpcService.db.QueryContext(ctx, "SELECT id, lock_name, held_by, held_until FROM long_lived_locks LIMIT 1000")
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	locks, err := vpcService.readLocks(rows)
+	if err != nil {
+		return nil, err
+	}
+
+	return &vpcapi.GetLocksResponse{Locks: locks}, nil
+}
+
+func (vpcService *vpcService) GetLock(ctx context.Context, id *vpcapi.LockId) (*vpcapi.Lock, error) {
+	rows, err := vpcService.db.QueryContext(ctx, "SELECT id, lock_name, held_by, held_until FROM long_lived_locks WHERE id = $1", id.GetId())
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	locks, err := vpcService.readLocks(rows)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(locks) == 0 {
+		return nil, status.Error(codes.NotFound, "lock not found")
+	}
+
+	return locks[0], nil
+}
+
+func (vpcService *vpcService) DeleteLock(ctx context.Context, id *vpcapi.LockId) (*empty.Empty, error) {
```

Comment:
So, is the intent behind this function to "unlock" the lock? It might make sense to build an actual mechanism to do that. 

Right now, the way you "knock" a lock holder out is by setting held_by to null, because that'll cause this to fail:

```
	result, err := tx.ExecContext(ctx, "UPDATE long_lived_locks SET held_until = now() + ($1 * interval '1 sec') WHERE id = $2 AND held_by = $3", lockTime.Seconds(), id, hostname)
	if err != nil {
		err = errors.Wrap(err, "Could update lock time")
		span.SetStatus(traceStatusFromError(err))
		return err
	}
	n, err := result.RowsAffected()
	if err != nil {
		err = errors.Wrap(err, "Could not get rows affected")
		span.SetStatus(traceStatusFromError(err))
		return err
	}
```

Reply:
Yep, totally. I added PreemptLock to do that. I was aiming for optionally allowing a lock to also be assigned to a new holder us that api but I don't think that's possible based on the acquire lock code. Please correct me if I'm wrong.